src/java.base/share/classes/java/util/random/RandomGeneratorFactory.java

```diff
@@ -0,0 +1,524 @@
+import java.util.random.RandomGenerator.LeapableGenerator;
+import java.util.random.RandomGenerator.SplittableGenerator;
+import java.util.random.RandomGenerator.StreamableGenerator;
+import java.util.ServiceLoader;
+import java.util.ServiceLoader.Provider;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import jdk.internal.util.random.RandomSupport.RandomGeneratorProperty;
+
+/**
+ * This is a factory class for generating multiple random number generators
+ * of a specific algorithm. {@link RandomGeneratorFactory} also provides
+ * methods for selecting random number generator algorithms.
+ *
+ * A specific {@link RandomGeneratorFactory} can be located by using the
+ * {@link RandomGenerator#factoryOf(String)} method, where the argument string
+ * is the name of the algorithm required. The method
+ * {@link RandomGenerator#all()} produces a {@link Stream} of all available
+ * {@link RandomGenerator RandomGenerators} that can be searched to locate a
+ * {@link RandomGeneratorFactory} suitable to the task.
+ *
+ * There are three methods for constructing a RandomGenerator instance,
+ * depending on the type of initial seed required.
+ * {@link RandomGeneratorFactory#create(long)} is used for long
+ * seed construction,
+ * {@link RandomGeneratorFactory#create(byte[])} is used for byte[]
+ * seed construction, and
+ * {@link RandomGeneratorFactory#create()} is used for random seed
+ * construction. Example;
+ *
+ * <pre>{@code
+ *    RandomGeneratorFactory<RandomGenerator> factory = RandomGenerator.factoryOf("Random");
+ *
+ *     for (int i = 0; i < 10; i++) {
+ *         new Thread(() -> {
+ *             RandomGenerator random = factory.create(100L);
+ *             System.out.println(random.nextDouble());
+ *         }).start();
+ *     }
+ * }</pre>
+ *
+ * RandomGeneratorFactory also provides methods describing the attributes (or properties)
+ * of a generator and can be used to select random number generator algorithms.
+ * These methods are typically used in
+ * conjunction with {@link RandomGenerator#all()}. In this example, the code locates the
+ * {@link RandomGeneratorFactory} that produces {@link RandomGenerator RandomGenerators}
+ * with the highest number of state bits.
+ *
+ * <pre>{@code
+ *     RandomGeneratorFactory<RandomGenerator> best = RandomGenerator.all()
+ *         .sorted((f, g) -> Integer.compare(g.stateBits(), f.stateBits()))
+ *         .findFirst()
+ *         .orElse(RandomGenerator.factoryOf("Random"));
+ *     System.out.println(best.name() + " in " + best.group() + " was selected");
+ *
+ *     RandomGenerator rng = best.create();
+ *     System.out.println(rng.nextLong());
+ * }</pre>
+ *
+ * @since   16
+ *
+ * @see java.util.random
+ *
+ */
+public final class RandomGeneratorFactory<T extends RandomGenerator> {
+    /**
+     * Map of provider classes.
+     */
+    private static volatile Map<String, Provider<? extends RandomGenerator>> factoryMap;
```

Comment:
should be FACTORY_MAP given that it's a static field

Reply:
will fall out of using  class holder idiom

Reply:
IntelliJ warns that this volatile field is unused. It has been replaced by the method getFactoryMap().